### PR TITLE
Feature/clean cmd

### PR DESCRIPTION
**Pull request by hidaruma**
I added the functions `-c(--clean)` and `-C(--clobber)` cf. #4 .
Unfortunately, I couldn't code rake test the functions because of my lack of rake's way. 

And moved `check_filename` function in runner to  in util.

`-c` removes files in array of `clean_files`, and `-C` removes in `clobber_files`.

In default, `clean_files` has

- `%B.aux`
- `%B.log`
- `%B.out` (hyperref package)
- `%B.blg` , `%B.bbl` , `%B.bcf`  and `%B-blx.bib` (BibTeX, BibLaTeX) 
- `%B.toc` (tableofcontents)
- `%B.fls`
- `%B.idx` and `%B.ilg` (makeindex)
- `%B.run.xml`

`clobber_files` has

- `%B.pdf`
- `%B.dvi`
- `%B.ps`
- `%B.synctex.gz`

### CODE REVIEW

**Comment by wtsnjp** [comment]
Thank you for pull requesting a nice feature! I really appreciate your contribution.

> Unfortunately, I couldn't code rake test the functions because of my lack of rake's way.

That's ok. I will write some tests for the feature later.

> `-C` removes in `clobber_files`

Just to confirm: the clobber action (`-C`, `--clobber`) removes all files specified in **`clean_files` and `clobber_files`**, right?

The default settings for `clean_files` and `clobber_files` looks reasonable to me. Great.

File: README.md [hunk 1]
```diff
@@ -150,9 +150,9 @@ This is the list of currently available TOML keys.
 		* `postprocess` (type: *string*)
 * `source` (type: *string* or *array of strings*, only for `llmk.toml`)
 * `max_repeat` (type: *int*, default: 3)
-* `clean_files` (type: *string* or *array of strings*, default: `["%B.aux", "%B.log", "%B.toc", "%B.out", "%B.bbl", "%B.bcf", "%B.blg", "%B.idx", "%B.synctex.gz"]`)
-* `clbber_files` (type: **string) or *array of strings*, default: `["%B.pdf", "%B.dvi", "%B.ps"]`)
-* `del_dir` (type: *string*)
+* `clean_files` (type: *string* or *array of strings*, default: `["%B.aux", "%B.log", "%B.toc", "%B.out", "%B.bbl", "%B.bcf", "%B.blg", "%B.idx", "%B.fls"]`)
+* `clbber_files` (type: *string or *array of strings*, default: `["%B.pdf", "%B.dvi", "%B.ps", "%B.synctex.gz"]`)
```
Comment (wtsnjp) [inline README.md]: typo: `clbber_files` -> `clobber_files`

File: llmk.lua [hunk 2]
```diff
@@ -58,9 +58,8 @@ M.top_level_spec = {
   source = {'*[string]', nil},
   sequence = {'[string]', {'latex', 'bibtex', 'makeindex', 'dvipdf'}},
   max_repeat = {'integer', 5},
-  clean_files ={'*[string]',{'%B.aux','%B.log', '%B.toc','%B.fls','%B.synctex.gz','%B.out', '%B.bbl', '%B.bcf', '%B.blg'}},
-  clobber_files ={'*[string]', {'%B.pdf', '%B.dvi', '%B.ps'}},
-  del_dir = {'string', nil},
+  clean_files ={'*[string]',{'%B.aux','%B.log', '%B.toc','%B.fls','%B.out', '%B.bbl', '%B.bcf', '%B.blg','%B-blx.bib', '%B.idx', '%B.ilg', '%B.run.xml'}},
+  clobber_files ={'*[string]', {'%B.pdf', '%B.dvi', '%B.ps', '%B.synctex.gz'}},
```
Comment (wtsnjp) [inline llmk.lua]: Please insert a single white space after every `=` and `,`

File: llmk.lua [hunk 3]
```diff
@@ -1128,140 +1133,104 @@ do -- The "cleaner" submodule
...
-    local filepath = (del_dir  .. tmp)
-    filepath = llmk.modify_path(filepath)
-    err = os.rename(tmp, filepath)
-    if err ~= nil then
-      llmk.dbg_print(err)
-    else
-      llmk.dbg_print(tmp "is moved to ".. del_dir .. ".\n") 
-    end
-      
+    llmk.util.err_print('info', fn .. ' is removed successfully.')
```
Comment (wtsnjp) [inline llmk.lua]: is -> has been

**Comment by hidaruma** [comment]
> Just to confirm: the clobber action (-C, --clobber) removes all files specified in clean_files and clobber_files, right?

Oh, that's right. `-C` removes files in `clean_files` first, and removes in `clobber_files` .

**Comment by wtsnjp** [comment]
> * `del_dir` : is string value. If set a diectory path to here, `-c` and `-C` options move the files into the directory set.

Oh, an explanation for `del_dir` remains in the README.md, but this feature is actually not implemented, right? If so, I will delete the line.

cf. https://github.com/wtsnjp/llmk/blame/a5aa3e543de1c901547ae81fb7255252128d3f50/README.md#L60-L63

**Comment by hidaruma** [comment]
Oops. I'm sorry, I forgot to delete the line `del_dir` . It is no problem that you delete the line.

> Oh, an explanation for del_dir remains in the README.md, but this feature is actually not implemented, right? If so, I will delete the line.

**Comment by wtsnjp** [comment]
Ok, I will take care of it. Thanks again :)

**Pull request merged**